core/src/test/java/org/elasticsearch/search/aggregations/metrics/HDRPercentileRanksIT.java

```diff
@@ -56,13 +58,66 @@
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.sameInstance;
 
-/**
- *
- */
-public class HDRPercentileRanksTests extends AbstractNumericTestCase {
+public class HDRPercentileRanksIT extends AbstractNumericTestCase {
+
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singleton(GroovyPlugin.class);
+        return Collections.singleton(CustomScriptPlugin.class);
+    }
+
+    public static class CustomScriptPlugin extends MockScriptPlugin {
+
+        @Override
+        protected Map<String, Function<Map<String, Object>, Object>> pluginScripts() {
+            Map<String, Function<Map<String, Object>, Object>> scripts = new HashMap<>();
+
+            scripts.put("20 - _value", vars -> {
+                double value = (double) vars.getOrDefault("_value", 0.0d);
+                return 20.0d - value;
+            });
+
+            scripts.put("_value - 1", vars -> {
+                double value = (double) vars.getOrDefault("_value", 0.0d);
+                return value - 1.0d;
+            });
+
+            scripts.put("_value - dec", vars -> {
+                double value = (double) vars.getOrDefault("_value", 0.0d);
+                int dec = (int) vars.get("dec");
+                return value - dec;
+            });
+
+            scripts.put("doc['value'].value", vars -> {
+                Map doc = (Map) vars.getOrDefault("doc", emptyMap());
+                return doc.get("value");
+            });
+
+            scripts.put("doc['value'].value - dec", vars -> {
+                int dec = (int) vars.get("dec");
+                Map doc = (Map) vars.getOrDefault("doc", emptyMap());
+                ScriptDocValues.Longs value = (ScriptDocValues.Longs) doc.get("value");
+                return value.getValue() - dec;
+            });
+
+            scripts.put("doc['values'].values", vars -> {
+                Map doc = (Map) vars.getOrDefault("doc", emptyMap());
```

Comment:
Can you add the `<String, Object>` (@nik9000 style) or `<?, ?>` (@jpountz style) to the map?

Again, I don't know if you should have a default here.